Jenkinsfile

```diff
@@ -48,129 +44,103 @@ pipeline {
+                                sh 'mvn install -B -pl $ITESTS -nsu'
                             }
                         }
-                    },
-                    windows: {
-                        node('proxmox-windows') {
-                            bat 'git config --system core.longpaths true'
-                            retry(3) {
-                                checkout scm
-                            }
-                            timeout(time: 3, unit: 'HOURS') {
-                                withMaven(maven: 'M35', jdk: 'jdk8-latest', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LARGE_MVN_OPTS}', options: [artifactsPublisher(disabled: true), dependenciesFingerprintPublisher(disabled: true, includeScopeCompile: false, includeScopeProvided: false, includeScopeRuntime: false, includeSnapshotVersions: false)]) {
-                                    bat 'mvn install -B -pl !%DOCS% -DskipStatic=true -DskipTests=true -T 1C'
-                                    bat 'mvn clean install -B -T 1C -pl !%ITESTS% -Dgib.enabled=true -Dgib.referenceBranch=/refs/remotes/origin/%CHANGE_TARGET%'
-                                    bat 'mvn install -B -pl %ITESTS% -nsu'
-                                }
+                    }
+                }
+                stage ('Windows') {
+                    agent { label 'server-2016-large' }
+                    steps {
+                        timeout(time: 3, unit: 'HOURS') {
+                            withMaven(maven: 'M35', jdk: 'jdk8-latest', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LARGE_MVN_OPTS}', options: [artifactsPublisher(disabled: true), dependenciesFingerprintPublisher(disabled: true, includeScopeCompile: false, includeScopeProvided: false, includeScopeRuntime: false, includeSnapshotVersions: false)]) {
+                                bat 'mvn install -B -pl !%DOCS% -DskipStatic=true -DskipTests=true -T 1C'
+                                bat 'mvn clean install -B -T 1C -pl !%ITESTS% -Dgib.enabled=true -Dgib.referenceBranch=/refs/remotes/origin/%CHANGE_TARGET%'
+                                bat 'mvn install -B -pl %ITESTS% -nsu'
                             }
                         }
                     }
-                )
+                }
             }
         }
         // The full build will be run against all regular branches
         stage('Full Build') {
             when { expression { env.CHANGE_ID == null } }
-            // TODO DDF-2971 refactor this stage from scripted syntax to declarative syntax to match the rest of the stages - https://issues.jenkins-ci.org/browse/JENKINS-41334
-            steps{
-                parallel(
-                    linux: {
-                        node('linux-large') {
-                            retry(3) {
```

Comment:
@mcalcote are we losing the retry functionality by switching away from the scripted syntax? Is there a way we can advise this still?

Reply:
Adding the retry logic back into the top level agent as well as each stage where we use an alternate agent.